arch/arc/include/swap_macros.h

```diff
@@ -412,12 +412,15 @@
 .macro _store_old_thread_callee_regs
 
 	_save_callee_saved_regs
-#ifdef CONFIG_SMP
-	/* save old thread into switch handle which is required by
-	 * wait_for_switch
+	/* Save old thread into switch handle which is required by wait_for_switch.
+	 * NOTE: we shouldn't save anything related to old thread context after this point!
+	 * TODO: we should add SMP write-after-write data memory barrier here, as we want all
```

Comment:
Why leave it as 'todo'? Create an issue for it to not forget?
(or, is the 'todo' intended as a possible future optimization/analysis task and not a problem in practice?)

Reply:
Because we don't have proper abstraction for ASM SMP barrier right now. We can't just put here an ASM instruction - it need to be different for EM / HS and UP / SMP cases.

This TODO represents only theoretical issue, so I've postponed it until interrupt code rework.